cpp/src/arrow/type.cc

```diff
@@ -2090,17 +2090,39 @@ Status SchemaBuilder::AreCompatible(const std::vector<std::shared_ptr<Schema>>&
   return Merge(schemas, policy).status();
 }
 
+std::vector<std::shared_ptr<Field>> make_fields(
+    std::initializer_list<std::pair<std::string, std::shared_ptr<DataType>>> init_list) {
```

Comment:
Can you please put this helper function in the anonymous namespace, and also make the function name CamelCase (i.e. `MakeFields`)?

Reply:
absolutely!